mediator/med_map_mod.F90

```diff
@@ -401,18 +403,19 @@ subroutine med_map_routehandles_initfrom_field(n1, n2, fldsrc, flddst, mapindex,
         polemethod = ESMF_POLEMETHOD_NONE ! todo: remove this when ESMF tripolar mapping fix is in place.
       endif
     else if (coupling_mode(1:4) == 'nems') then
-       if (n1 == compatm .and. (n2 == compocn .or. n2 == compice)) then
+       if ( (n1 == compocn .or. n1 == compice .or. n1 == compwav) .and. &
```

Comment:
This part is getting more complex and confusing. I think we need to find way to make it simpler in the near future.

Reply:
@uturuncoglu @DeniseWorthen  - I agree that we need to refactor this part of the code. Can we meet early next week to come up with a plan. The sooner we do this the better.

Reply:
Yes, I agree.